bots/tools/code_tools.py

```
@@ -23,20 +23,111 @@ def _write_file_bom_safe(file_path: str, content: str) -> None:
+                    if file.lower().endswith(tuple(extensions_list_lower)):
+                        dir_entries.append((level + 1, f"{subindent}{file}"))
+                        max_level = max(max_level, level + 1)
+
+        # Second pass: apply length limiting by progressively removing deeper levels
+        current_level_limit = max_level
+        while current_level_limit >= 0:
+            # Filter entries up to current level limit
+            filtered_entries = [(level, line) for level, line in dir_entries if level <= current_level_limit]
+            # Check if we need to add truncation indicators
+            truncated_dirs = set()
+            for level, line in dir_entries:
```

Ruff (0.15.9)

[warning] Loop control variable `line` not used within loop body

(B007)